pytest/tests/mocknet/sharded_bm.py

```diff
@@ -0,0 +1,338 @@
+"""
+This script is used to run a sharded benchmark on a forknet.
+"""
+
+from argparse import ArgumentParser
+import os
+import sys
+import json
+import copy
+import pathlib
+import subprocess
+
+import time
+from types import SimpleNamespace
+from mirror import CommandContext, get_nodes_status, init_cmd, new_test_cmd, \
+    reset_cmd, run_env_cmd, run_remote_cmd, run_remote_upload_file, \
+    start_nodes_cmd, stop_nodes_cmd, update_binaries_cmd
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+from configured_logger import logger
+
+# cspell:words BENCHNET
+
+# TODO: consider moving local directory to pytest.
+LOCAL_BENCHNET_DIR = "../benchmarks/sharded-bm"
+
+REMOTE_BENCHNET_DIR = "/home/ubuntu/bench"
+NEAR_HOME = "/home/ubuntu/.near"
+CONFIG_PATH = f"{NEAR_HOME}/config.json"
+
+
+def fetch_forknet_details(forknet_name):
+    """Fetch the forknet details from GCP."""
+    find_instances_cmd = [
+        "gcloud", "compute", "instances", "list", "--project=nearone-mocknet",
+        f"--filter=name~'-{forknet_name}-' AND -name~'traffic' AND -name~'tracing'",
+        "--format=get(name,networkInterfaces[0].networkIP)"
+    ]
```

Comment:
I guess this works but I believe there is a gcp library that you can use from python. It may save us some issues in parsing outputs from different versions of the cli, handling upgrades and would generally be more type safe and robust. It's fine as a first step though.